backend/app/api/endpoints/export.py

```
@@ -341,6 +342,104 @@ async def export_book_epub(
+    except ExportTimeoutError as e:
+        logger.warning("Markdown export timed out for book %s", book_id)
+        raise HTTPException(status_code=504, detail=str(e)) from e
+    except ExportUnavailableError as e:
+        logger.warning("Markdown export unavailable: %s", e)
+        raise HTTPException(status_code=503, detail=str(e)) from e
+    except Exception:
+        logger.error("Failed to generate Markdown", exc_info=True)
+        raise HTTPException(
+            status_code=500,
+            detail="Failed to generate Markdown"
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)